### DISCUSSION TITLE
Segmented control

### DISCUSSION THREAD

**Discussion by manfromarce** [CommunityToolkit/Maui#2335] [New Feature Discussions]
It would be nice to have a segmented control to quickly select an option or switch views. A native equivalent should be available in all platforms.
Material: [https://m3.material.io/components/segmented-buttons/overview](https://m3.material.io/components/segmented-buttons/overview)
Apple: [https://developer.apple.com/design/human-interface-guidelines/segmented-controls](https://developer.apple.com/design/human-interface-guidelines/segmented-controls)
WinUI Community Toolkit: [https://github.com/CommunityToolkit/Windows/tree/main/components/Segmented/src](https://github.com/CommunityToolkit/Windows/tree/main/components/Segmented/src)
There are also third-party implementations for MAUI.

**Comment by stephenquan** [comment]
Here's a mockup of how you may achieve a SegementedControl with a CollectionView:

```xml
<CollectionView x:Name="segmentedControl" HorizontalOptions="Fill" SelectionMode="Single" VerticalOptions="Start">
    <CollectionView.ItemsSource>
        <x:Array Type="{x:Type x:String}">
            <x:String>Alpha</x:String>
            <x:String>Beta</x:String>
            <x:String>Gamma</x:String>
            <x:String>Delta</x:String>
        </x:Array>
    </CollectionView.ItemsSource>
    <CollectionView.ItemsLayout>
        <GridItemsLayout Orientation="Vertical" Span="4" />
    </CollectionView.ItemsLayout>
    <CollectionView.ItemTemplate>
        <DataTemplate x:DataType="x:String">
            <Frame>
                <Label HorizontalOptions="Center" Text="{Binding .}" />
                <VisualStateManager.VisualStateGroups>
                    <VisualStateGroupList>
                        <VisualStateGroup Name="CommonStates">
                            <VisualState Name="Normal">
                                <VisualState.Setters>
                                    <Setter Property="BackgroundColor" Value="White" />
                                </VisualState.Setters>
                            </VisualState>
                            <VisualState Name="Selected">
                                <VisualState.Setters>
                                    <Setter Property="BackgroundColor" Value="LightBlue" />
                                </VisualState.Setters>
                            </VisualState>
                        </VisualStateGroup>
                    </VisualStateGroupList>
                </VisualStateManager.VisualStateGroups>
            </Frame>
        </DataTemplate>
    </CollectionView.ItemTemplate>
</CollectionView>
```

![SegmentedControl](https://github.com/user-attachments/assets/f3227b71-f8f0-41ed-a2c8-39a860a52b81)

**Reply by MichaelLHerman** [reply]
SyncFusion's MAUI toolkit also includes a Segmented Control, but it would be nice to have one that surfaces the platforms' controls and doesn't draw its own UI 

https://help.syncfusion.com/maui/segmented-control/overview
https://github.com/syncfusion/maui-toolkit/tree/main/maui/src/SegmentedControl
